docs/cookbook/makeMockTemporal.mjs

```diff
@@ -0,0 +1,270 @@
+// First of all, create a controllable clock object that underlies the
+// functions in the Temporal.now namespace, that we can tick forward or backward
+// at will.
+// We'll use the clock to remove Temporal's access to the system clock below.
+
+class Clock {
+  epochNs = 0n;
+  tick(ticks = 1) {
+    this.epochNs += BigInt(ticks);
+  }
+}
+const clock = new Clock();
+
+// Save the original Temporal functions that we will override but still need
+// access to internally.
+
+const realTemporalCalendar = Temporal.Calendar;
+const realCalendarFrom = Temporal.Calendar.from;
+const realTemporalTimeZone = Temporal.TimeZone;
+const realTimeZoneFrom = Temporal.TimeZone.from;
+const realTemporalNow = Temporal.now;
+
+// Override the Temporal.Calendar constructor and Temporal.Calendar.from to
+// disallow all calendars except the iso8601 calendar, otherwise insecure code
+// might be able to tell something about the version of the host system's
+// locale data.
```

Comment:
The version of the locale data present on the host system is a browser fingerprinting vector. If you can construct dates using non-ISO calendars and convert them to ISO, then you may be able to tell (due to bugs being fixed, or adjustments to calendars based on astronomical sightings) which version of locale data is present.

Reply:
i see, so this example is primarily about avoiding fingerprinting? that's usually not something a web site author wants to prevent, since they generally want to use that data for analytics.

Reply:
This example is a grab bag of modifications that are somewhat related. The blurb I wrote in cookbook-mock.md says that you wouldn't need all of them for every purpose. For example, for a test harness, you wouldn't bother with blocking access to the time zones and calendars, and for an environment like SES, you wouldn't bother with the controllable clock.

Reply:
gotcha.

this is way better than nothing, to be sure :-) but it might be less confusing as separate examples, targeted to a single use case.

Reply:
Fair enough. I'll see what would be the best way to split it up after getting comments from others.

Reply:
I think comments like this serve the purpose of pointing out the use case of / reasons for each part - no need to further split up imo as there aren't that many things it locks down (clock precision & time, locale). Perhaps adding a bit more to the markdown intro about preventing clock & locale fingerprinting would be sufficient.